qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/BackupMultiPartitionTest.java

```diff
@@ -0,0 +1,321 @@
+    backupConfig.setStore(BackupStoreType.S3);
+
+    final var s3Config = backupConfig.getS3();
+
+    generateBucketName();
+
+    s3Config.setBucketName(bucketName);
+    s3Config.setEndpoint(S3.externalEndpoint());
+    s3Config.setRegion(S3.region());
+    s3Config.setAccessKey(S3.accessKey());
+    s3Config.setSecretKey(S3.secretKey());
+  }
+
+  private void generateBucketName() {
+    // Generate only once per test
+    if (bucketName == null) {
+      bucketName = RandomStringUtils.randomAlphabetic(10).toLowerCase();
+    }
+  }
+
+  void createBackupStoreForTest() {
+    // Create bucket before for storing backups
+    s3ClientConfig =
+        S3BackupConfig.from(
+            bucketName, S3.externalEndpoint(), S3.region(), S3.accessKey(), S3.secretKey());
+    s3BackupStore = new S3BackupStore(s3ClientConfig);
+    try (final var s3Client = S3BackupStore.buildClient(s3ClientConfig)) {
+      s3Client.createBucket(builder -> builder.bucket(bucketName).build()).join();
+    }
+  }
+
+  @BeforeEach
+  void setup() {
+    client = new GrpcClientRule(clusteringRule.getClient());
+    backupRequestHandler = new BackupRequestHandler(clusteringRule.getGateway().getBrokerClient());
+    createBackupStoreForTest();
+  }
+
+  @AfterEach
+  void close() {
+    // Create bucket before for storing backups
+    s3BackupStore.closeAsync();
+    // reset so that each test can use a different bucket name
+    bucketName = null;
+  }
+
+  @Test
+  @Timeout(value = 120)
+  void shouldTriggerBackupViaInterPartitionMessageDeploymentDistribute() {
+    // given
+    final long backupId = 1;
+    // trigger backup only on partition 1
+    takeBackupOnPartition(backupId, 1);
+
+    // when
+    client.deployProcess(SIMPLE_PROCESS);
+
+    // then
+    waitUntilBackupIsCompleted(backupId);
+  }
+
+  @Test
+  @Timeout(value = 120)
+  void shouldTriggerBackupViaInterPartitionMessageSubscriptionCommands() {
+    // given
+    final long processKey = client.deployProcess(PROCESS_WITH_MESSAGE_EVENT);
+
+    final long backupId = 2;
+    // trigger backup only on partition 1
+    takeBackupOnPartition(backupId, 1);
+
+    // when
+    createProcessInstanceOnPartitionOne(processKey);
+
+    // then
+    waitUntilBackupIsCompleted(backupId);
+  }
+
+  @Test
+  @Timeout(value = 120)
+  void shouldTriggerBackupViaInterPartitionMessageCorrelationCommands() {
+    // given
+    final long processKey = client.deployProcess(PROCESS_WITH_MESSAGE_EVENT);
+    createProcessInstanceOnPartitionOne(processKey);
+
+    final long backupId = 3;
+    // trigger backup only on partition 2
+    takeBackupOnPartition(backupId, 2);
+
+    // when
+    publishMessageAndWaitUntilCorrelated();
+
+    // then
+    waitUntilBackupIsCompleted(backupId);
+  }
+
+  @Test
+  @Timeout(value = 180)
+  void shouldRestoreOnAllPartitions() throws InterruptedException {
+    // given
+    // create instances on all partitions.
+    client.createJobs(JOB_TYPE, clusteringRule.getPartitionCount());
```

Comment:
:x: Unless I'm missing something, this is not correct. `createJobs` reads as if it just creates a fixed amount of jobs without making guarantees about the distribution over partitions.

Reply:
The assumption here is that gateway distributes the request in a round-robin way so all partitions receive one request each. But you are right. It does not guarantee it. I will update this.